crates/unigen/src/builder/core/neutron.rs

```diff
@@ -9,19 +9,21 @@ pub struct Neutrons {
 impl Neutrons {
     pub fn new(count: u32) -> Self {
         let mut neutrons = [NeutronData::Unknown; 118];
+        neutrons[0..(count as usize)]
+            .iter_mut()
+            .for_each(|n| *n = NeutronData::new(Neutron::new()));
 
+        /*
```

Comment:
is this slower than iter_mut? Sometimes single threaded wins. It depends 🙃

Reply:
`iter_mut` is single threaded you might be thinking of rayon's `par_iter_mut` 
but anyway from what i found from testing on rust playground using `iter_mut` seems to be able to do an inline memcpy compared to directly indexing which compiles down to a simple move compare and jump

Reply:
I just ran a test and using `iter_mut` performs maybe a few percent faster but not really noticeable.
If you don't like to `for_each` you could replace it with something like this
```
for x in &mut arr[0..end]{
    *x = value;
}
```

Reply:
Ok makes sense. I don't mind either approach. I guess I just saw `iter_mut` and assumed `par`.

Reply:
In this case you can get rid of the commented out for blocks. I like it.